drivers/ethernet/eth_xilinx_axienet.c

```diff
@@ -0,0 +1,570 @@
+/*
+ * Xilinx AXI 1G / 2.5G Ethernet Subsystem
+ *
+ * Copyright(c) 2024, CISPA Helmholtz Center for Information Security
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(eth_xilinx_axienet, CONFIG_ETHERNET_LOG_LEVEL);
+
+#include <sys/types.h>
+#include <zephyr/kernel.h>
+#include <zephyr/net/ethernet.h>
+#include <ethernet/eth_stats.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/barrier.h>
+
+#include "../dma/dma_xilinx_axi_dma.h"
+
+/* register offsets and masks */
+#define XILINX_AXIENET_INTERRUPT_STATUS_OFFSET         0x0000000C
+#define XILINX_AXIENET_INTERRUPT_STATUS_RXREJ_MASK     0x00000008
+#define XILINX_AXIENET_INTERRUPT_STATUS_RXFIFOOVR_MASK 0x00000010 /* Rx fifo overrun */
+#define XILINX_AXIENET_INTERRUPT_PENDING_OFFSET        0x00000010
+
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXCMPIT_MASK     0x00000004 /* Rx complete */
+#define XILINX_AXIENET_INTERRUPT_PENDING_RXRJECT_MASK     0x00000008 /* Rx frame rejected */
```

Comment:
it appears to be duplicate macros,
XILINX_AXIENET_INTERRUPT_PENDING_RXRJECT_MASK and
XILINX_AXIENET_INTERRUPT_PENDING_RXREJ_MASK.

Reply:
These refer to two bits in two subsequent registers, that just happen to be in the same place. Here is the reference: https://docs.amd.com/r/en-US/pg138-axi-ethernet/Register-Space
The macros refer to the RXREJ bit in the Interrupt Status and Interrupt Pending registers, respectively.

Reply:
RxRject bit is present in **interrupt status**, **interrupt pending** and **interrupt Enable** registers.

RxRject bit in **interrupt status register** was identified by below macro:
#define XILINX_AXIENET_INTERRUPT_STATUS_RXREJ_MASK     0x00000008

RxRject bit in **interrupt enable register** was identified by below macro:
#define XILINX_AXIENET_INTERRUPT_ENABLE_RXREJ_MASK  0x00000008

RxRject bit in **interrupt pending register** was identified by below two macro's, which appears to be duplicate:
#define XILINX_AXIENET_INTERRUPT_PENDING_RXRJECT_MASK     0x00000008
#define XILINX_AXIENET_INTERRUPT_PENDING_RXREJ_MASK 0x00000008

Reply:
You are right, I missed that. I have removed the duplicate.